src/__tests__/search-and-replace-test.jsx

```diff
@@ -0,0 +1,185 @@
+var TestUtils = React.addons.TestUtils;
+
+var itemJson = require("./test-items/search-and-replace-item.json");
+var particleJson = require("./test-items/search-and-replace-particle.json");
+
+describe("Search and Replace", function() {
+    var editor, editorProps;
+    var searchInput, replaceInput;
+    var previousButton, nextButton, replaceButton, replaceAllButton;
+
+    beforeEach(function () {
+        var problemNum = Math.floor(Math.random() * 100);
+        var enabledFeatures = {
+            highlight: true,
+            toolTipFormats: true,
+            useMathQuill: true
+        };
+
+        editorProps = {
+            problemNum: problemNum,
+            enabledFeatures: enabledFeatures,
+            developerMode: true,
+            imageUploader: function(image, callback) {
+                _.delay(callback, 1000, "http://fake.image.url");
+            },
+            apiOptions: {
+                fancyDropdowns: true,
+                __onInputError: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["onInputError:"].concat(args));
+                    return true;
+                },
+                __interceptInputFocus: function() {
+                    var args = _.toArray(arguments);
+                    console.log.apply(console, ["interceptInputFocus:"].concat(args));
+                    return;
+                },
+                onFocusChange: function(newPath, oldPath) {
+                    console.log("onFocusChange", newPath, oldPath);
+                },
+                __staticRender: true
+            }
+        };
+    });
+
+    describe("EditorPage", function() {
+        beforeEach(function() {
+            editorProps = $.extend(true, editorProps, itemJson);
+            editorProps.componentClass = EditorPage;
+
+            var StatefulEditorPageFactory = React.createFactory(StatefulEditorPage);
+            editor = React.render(
+                StatefulEditorPageFactory(editorProps, null),
+                document.body
+            );
+
+            var dialog = editor.refs.searchAndReplace;
+
+            searchInput = dialog.refs.searchInput.getDOMNode();
+            replaceInput = dialog.refs.replaceInput.getDOMNode();
+            replaceButton = dialog.refs.replaceButton.getDOMNode();
+            replaceAllButton = dialog.refs.replaceAllButton.getDOMNode();
+            previousButton = dialog.refs.previousButton.getDOMNode();
+            nextButton = dialog.refs.nextButton.getDOMNode();
+        });
+
+        afterEach(function (done) {
+            React.unmountComponentAtNode(document.body);
+            document.body.innerHTML = "";
+            setTimeout(done);
+        });
+
+        it("should replace all instances of the searchString not inside a widget reference", function() {
+            React.addons.TestUtils.Simulate.change(searchInput, {target: {value: "categorizer"}});
```

Comment:
Well it looks like this is more broked than i thought it was. There's a lint_blacklist.txt in perseus' root directory that says to ignore everything in react-components, but then react-components must have added a lint_blacklist.txt itself since I added that, which doesn't handle all the necessary cases, so we get a bunch of errors from the examples and docs folders (among others).

I've blacklisted a few things on master of perseus and react-components, which brings the number of errors down from "insane" to "very embarrassing".

Obviously we don't have the best setup for this (but we'll try not to make it too much worse).